node/node.go

```diff
@@ -113,10 +112,8 @@ func (n *Node) mempoolPublishLoop(ctx context.Context) {
 		// send transactions
 		for {
 			n.Logger.Debug("Gossiping...")
-			// TODO(tzdybal): refactor lazyledger-core to avoid reflection
-			v := reflect.Indirect(reflect.ValueOf(next.Value))
-			f := v.FieldByName("tx")
-			tx := f.Bytes()
+			memTx := next.Value.(*mempool.MempoolTx)
```

Comment:
What happens if the cast fails?

Reply:
We will see a panic in tests and will be able to fix the issue.
Initially I created idiomatic error handling for this cast... But in this particular case I think we should just panic.

Reply:
I see, I briefly thought this was the other way around: you could send a Tx and crash the process / node but in this case the process is in control.